src/status_im2/contexts/communities/overview/view.cljs

```diff
@@ -159,11 +157,14 @@
 
 (def channel-list-component (memoize channel-list-component-fn))
 
-(defn render-page-content [{:keys [name description locked joined id images
-                                   status tokens tags community-color]
+(def request-to-join-text (memoize (fn [is-open?] (if is-open? (i18n/label :t/join-open-community) (i18n/label :t/request-to-join-community)))))
+
+(defn render-page-content [{:keys [name description locked joined id images can-join?
+                                   status tokens tags community-color permissions]
                             :as   community}
                            channel-heights first-channel-height]
-  (let [thumbnail-image (get-in images [:thumbnail])]
+  (let [is-open? (not= 3 (:access permissions))
```

Comment:
What is this [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) `3`? It looks like it should be refactored into a constant.

Reply:
resolved above